lib/l2.c

```diff
@@ -101,18 +102,29 @@ verify_l2_hdr(struct gatekeeper_if *iface, struct rte_ether_hdr *eth_hdr,
 			return -1;
 		}
 
+		if (strcmp(proto_name, "ARP") == 0)
+			vlan_tag_be = iface->ipv4_vlan_tag_be;
+		else if (strcmp(proto_name, "ND") == 0)
+			vlan_tag_be = iface->ipv6_vlan_tag_be;
+		else {
+			G_LOG(WARNING,
+				"l2: %s interface incorrectly received a packet with unknown protocol %s\n",
+				iface->name, proto_name);
+			return -1;
+		}
```

Comment:
Variable `proto_name` is meant to improve log entries. Moreover, `verify_l2_hdr()` is only called twice. It's better to add parameter `uint16_t vlan_tag_be` to `verify_l2_hdr()` and get this information from callers.